File: Admin_p/Admin_p.Web/Modules/ServerTypes/electroDB/PromotionsColumns.ts
import { ColumnsBase, fieldsProxy } from "@serenity-is/corelib";
import { Column } from "@serenity-is/sleekgrid";
import { PromotionsRow } from "./PromotionsRow";

export interface PromotionsColumns {
    PromotionId: Column<PromotionsRow>;
    Name: Column<PromotionsRow>;
    Description: Column<PromotionsRow>;
    DiscountPercentage: Column<PromotionsRow>;
    StartDate: Column<PromotionsRow>;
    EndDate: Column<PromotionsRow>;
    IsActive: Column<PromotionsRow>;
    CreatedAt: Column<PromotionsRow>;
    UpdatedAt: Column<PromotionsRow>;
}

export class PromotionsColumns extends ColumnsBase<PromotionsRow> {
    static readonly columnsKey = 'electroDB.Promotions';
    static readonly Fields = fieldsProxy<PromotionsColumns>();
}

File: Admin_p/Admin_p.Web/Modules/electroDB/Promotions/PromotionsColumns.cs
using Serenity.ComponentModel;
using System;
using System.ComponentModel;

namespace Admin_p.electroDB.Columns;

[ColumnsScript("electroDB.Promotions")]
[BasedOnRow(typeof(PromotionsRow), CheckNames = true)]
public class PromotionsColumns
{
    [EditLink, DisplayName("Db.Shared.RecordId"), AlignRight]
    public int PromotionId { get; set; }
    [EditLink]
    public string Name { get; set; }
    public string Description { get; set; }
    public decimal DiscountPercentage { get; set; }
    public DateTime StartDate { get; set; }
    public DateTime EndDate { get; set; }
    public bool IsActive { get; set; }
    public DateTime CreatedAt { get; set; }
    public DateTime UpdatedAt { get; set; }
}